fix(datasets): consistent dataset list

zhaoyongjie opened the pull request:
### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
closes: https://github.com/apache/superset/issues/14262

Currently, between the `Dataset list view` and `change dataset modal` datasets order is inconsistent. The `change dataset modal` can not display all datasets. 

This PR make
1. `Dataset list view` and `change dataset modal` datasets order is consistent
2. make server pagination in `change dataset modal`
3. All `datasets` can be displayed in `change dataset modal`
4. backend sort by in `change dataset modal`
5. use `owners` replace `creator`


### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->
#### Before
<img width="1471" alt="image" src="https://user-images.githubusercontent.com/2016594/120994394-044c8b80-c7b7-11eb-9558-e19d7d8d425a.png">

#### After
Consistent dataset list between `dataset list` and `Change dataset modal`
<img width="1289" alt="image" src="https://user-images.githubusercontent.com/2016594/121183701-eeae9300-c896-11eb-9121-0699be40ea83.png">



https://user-images.githubusercontent.com/2016594/121185206-67fab580-c898-11eb-9f7c-5b0d61c79d9c.mp4





### TESTING INSTRUCTIONS
<!--- Required! What steps can be taken to manually verify the changes? -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [x] Has associated issue: https://github.com/apache/superset/issues/14262
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

zhaoyongjie committed:
make consistent dataset list

zhaoyongjie committed:
server pagination

junlincc commented:
/testenv up

junlincc commented:
😞thanks for the PR. 

1. after the PR, sorting no longer work~ :(  please make sure when we add new feature, the old related behavior is preserved. 
2. not related to this PR, please have pagination fixed at the bottom on scroll. 

https://user-images.githubusercontent.com/67837651/121073555-802edf80-c787-11eb-9697-a8a2e95c8573.mov

zhaoyongjie commented:
@junlincc 
1. The frontend sorting seems that not work previous. the after screenshot is on the master branch.
<img width="896" alt="image" src="https://user-images.githubusercontent.com/2016594/121109602-7e742300-c83e-11eb-8088-3755b7256c82.png">
2. the pagination flowing, I will fix it in separate PR.

zhaoyongjie committed:
make paginations sticky

zhaoyongjie committed:
server side sorting by

zhaoyongjie committed:
refactor

zhaoyongjie committed:
fix css

zhaoyongjie committed:
fix ci

zhaoyongjie commented:
/testenv up

zhaoyongjie committed:
fix ut

junlincc commented:
/testenv up

junlincc commented:
@zhaoyongjie thanks for addressing the issues! Pagination is 'fixed' and Sort works mostly as expected! 🙏
only thing I'm not sure is the cap F goes to the top of the list. 
I spot checked pagination in other areas, list view, data south table, they all look good! 

https://user-images.githubusercontent.com/67837651/121239710-6efcd600-c84e-11eb-899d-4c4d76b5ae9d.mov

zhaoyongjie commented:
> @zhaoyongjie thanks for addressing the issues! Pagination is 'fixed' and Sort works mostly as expected! 🙏
> only thing I'm not sure is the cap F goes to the top of the list.
> I spot checked pagination in other areas, list view, data south table, they all look good!
> Screen.Recording.2021-06-08.at.11.38.44.AM.mov

Due to server sorting, it's order is ASCII order. refer to:

https://support.ecisolutions.com/doc-ddms/help/reportsmenu/ascii_sort_order_chart.htm

villebro reviewed (approved):
This is really great work! Code LGTM; I tested all affected areas as best as I could and couldn't find any regressions. The sort problem @junlincc identified is indeed confusing, but I assume that's down to the backend database collation, so needs to be fixed elsewhere.

junlincc commented:
Got it, let's address it in a separate PR. 🟢

junlincc reviewed (approved):
product sign-off. remaining sorting known issue will be addressed in follow-up PRs.
